### PR TITLE
[SPARK-14894][PySpark] Add result summary api to Gaussian Mixture

### PR DESCRIPTION
**Pull request by GayathriMurali**
## What changes were proposed in this pull request?

Add summary API to Gaussian Mixture
## How was this patch tested?

Added unit test case to test summary information

### CODE REVIEW

**Comment by GayathriMurali** [comment]
@wangmiao1981 @jkbradley  Please help review this PR

File: python/pyspark/ml/clustering.py [hunk 1]
```diff
@@ -58,6 +58,125 @@ def gaussiansDF(self):
         """
         return self._call_java("gaussiansDF")
 
+    @property
+    @since("2.0.0")
+    def summary(self):
+        """
```
Comment (holdenk) [inline python/pyspark/ml/clustering.py]: This PyDoc seems oddly formatted in terms of line breaks.

**Comment by GayathriMurali** [comment]
@jkbradley Can you please ok to test this

**Comment by GayathriMurali** [comment]
@holdenk I fixed the pydoc style issue. Can you please help review this?

File: python/pyspark/ml/clustering.py [hunk 2]
```diff
@@ -58,6 +58,124 @@ def gaussiansDF(self):
...
+        Gets summary of model on training set. An exception is thrown if
+        `trainingSummary is None`.
```
Comment (holdenk) [inline python/pyspark/ml/clustering.py]: this also seems strangely formatted - did you mean `trainingSummary` is `None`. ?

File: python/pyspark/ml/clustering.py [hunk 3]
```diff
@@ -58,6 +58,124 @@ def gaussiansDF(self):
...
+    @since("2.0.0")
+    def hasSummary(self):
+        """
+        Indicates whether a training summary exists for this model
+        instance.
+        """
+        return self._call_java("hasSummary")
+
+    @since("2.0.0")
+    def evaluate(self, dataset):
```
Comment (holdenk) [inline python/pyspark/ml/clustering.py]: This function doesn't appear to be present in the Scala `GaussianMixtureModel` - can you double check your functions against the Scaladoc to make sure your wrapping the correct methods?

**Comment by holdenk** [comment]
So did a quick look and it seems like it might not be lining up with the Scala implementation (or I just haven't had enough coffee this morning). It might be worthwhile to doublecheck against the scaladoc and also run your tests locally if you haven't done that.

**Comment by GayathriMurali** [comment]
@holdenk I checked the ScalaDoc and removed the evaluate method. Thanks for pointing it out. Can you please help review

**Comment by GayathriMurali** [comment]
@jkbradley @holdenk Can you please help review?

**Comment by GayathriMurali** [comment]
@jkbradley This PR has been open >30days. Can you please help review?

**Comment by MLnick** [comment]
ok to test

File: python/pyspark/ml/tests.py [hunk 4]
```diff
@@ -1070,6 +1070,21 @@ def test_logistic_regression_summary(self):
         sameSummary = model.evaluate(df)
         self.assertAlmostEqual(sameSummary.areaUnderROC, s.areaUnderROC)
 
+    def test_gaussian_mixture_summary(self):
+        from pyspark.mllib.linalg import Vectors
+        df = self.spark.read.format("libsvm").load("data/mllib/sample_kmeans_data.txt")
```
Comment (GayathriMurali) [inline python/pyspark/ml/tests.py]: Please let me know if its ok to load data from a file when all other test cases uses hard coded data values. I tried with a sparse vector and fit gave me an error for the data format.

**Comment by GayathriMurali** [comment]
@MLnick It would be great if you can help review this.

File: python/pyspark/ml/tests.py (hunk 4 above)
Comment (holdenk) [inline python/pyspark/ml/tests.py]: I think its ok to load a dataset which is already used for testing else where.

File: python/pyspark/ml/clustering.py [hunk 5]
```diff
@@ -56,8 +56,83 @@ def gaussiansDF(self):
         """
         return self._call_java("gaussiansDF")
 
+    @property
+    @since("2.0.0")
```
Comment (holdenk) [inline python/pyspark/ml/clustering.py]: Since we've shipped 2.0 we will need to update this to 2.1 (same with the versionAdded notes)

**Review by holdenk — COMMENTED** [review]
Are you still interested in this PR @GayathriMurali  ? If you've got the time to look at some of my notes around the PyDocs that would be cool - but we can also try pinging @davies & @MLnick to see if they are available to review once those comments are looked at :)

**Comment by HyukjinKwon** [comment]
Hi @GayathriMurali, it seems inactive for the review comments for few months. Should this be better closed for now if you are not able to proceed this further?

**Comment by wangmiao1981** [comment]
@GayathriMurali If you are not able to proceed, I can take over. Thanks!

**Comment by wangmiao1981** [comment]
#15777 has resolved this issue. We should close this one.

**Comment by wangmiao1981** [comment]
@HyukjinKwon @srowen This should be closed. Thanks!

**Comment by HyukjinKwon** [comment]
@wangmiao1981 Wow, thank you for your efforts to verify it!

**Pull request closed without merging**